torch/jit/_recursive.py

```diff
@@ -88,6 +90,17 @@ def infer_raw_concrete_type(nn_module):
         added_names.add(name)
 
     for name, item in nn_module._modules.items():
+        if name in class_annotations:
```

Comment:
This logic needs to be carefully done to avoid being fragile. First, it should avoid using continue. Because of continue it forgets to add name to `added_names`, which is probably a bug. The handling of the submodule is also weird. There is the concrete submodule, which needs to be converted into script using the get_or_create_concrete_type, and there is the interface type which needs to be added as an attribute. This seems to skip handling the concrete_type for the actual module being stored inside this submodule. I think this leads to the problem later where all submodules get recursively scripted again.